### PR TITLE
add boilerplate

### PR DESCRIPTION
**Pull request by gnegri**

### CODE REVIEW

File: common/config.ini.template [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[agent]
+# define things to include (default all)
+thing1_filter =
```
Comment (ptmr3) [inline common/config.ini.template]: please rename thing1 and thing2

File: common/insightagent-boilerplate.py [hunk 2]
```diff
@@ -0,0 +1,587 @@
...
+import httplib
+import requests
+
+'''
+This script gathers data to send to Insightfinder
+'''
+
+
+def start_data_processing():
+    """ TODO: replace with your code.
```
Comment (ptmr3) [inline common/insightagent-boilerplate.py]: This could be a good place to allow a modular processing step (check for an ancillary parser file and use to process the data if present)

File: common/insightagent-boilerplate.py [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+    track['timestamp_format'] = 'See datetime.strptime documentation'
+    
+
+def get_agent_config_vars():
+    """ Read and parse config.ini """
+    if os.path.exists(os.path.abspath(os.path.join(__file__, os.pardir, 'config.ini'))):
+        config_parser = ConfigParser.SafeConfigParser()
+        config_parser.read(os.path.abspath(os.path.join(__file__, os.pardir, 'config.ini')))
+        try:
+            ## TODO: fill out the fields to grab. examples given below
```
Comment (ptmr3) [inline common/insightagent-boilerplate.py]: What is to be pulled from a config could be passed into this function. If this file is to become an imported module, we need to make this standalone, not requiring any changes for specific data types

File: common/insightagent-boilerplate.py (hunk 2 above)
Comment (gnegri) [reply common/insightagent-boilerplate.py]: Agreed, but due to the current design (copy it and edit from there) vs an importable module, it would be awkward to change where the agent config vars are grabbed. I want to get to an importable module where we could be more thoughtful about the design from the get-go; this was 90% written while re-writing things for the Zipkin agent, so it was just restructuring and formalizing what we were doing for agents before.

In short, yes, but IMO it's out of scope.

File: common/insightagent-boilerplate.py (hunk 3 above)
Comment (gnegri) [reply common/insightagent-boilerplate.py]: see previous comment.
